app/controllers/scim_rails/scim_users_controller.rb

```diff
@@ -115,11 +116,48 @@ def delete
 
     private
 
+    # `process_path` is a method that parses the string in the "path"
+    # key of a PATCH operation. Together with the "value" key, it
+    # converts it into a Hash that can be used in the `permitted_params`
+    # method to help update the attributes of a User.
+    #
+    # Example: given the following operation:
+    #   operation = {
+    #     'op': 'Replace',
+    #     'path': 'name.givenName',
+    #     'value': 'Grayson'
+    #   }
+    # calling `process_path(operation)` will return the Hash:
+    #   {
+    #     name: {
+    #       givenName: 'Grayson'
+    #     }
+    #   }
+    # which can easily be processed by `permitted_params` which will get
+    # the attributes that will be updated by the PATCH request
+    def process_path(operation)
+      keys = operation["path"].split('.').map { |key| key.to_sym }
+
+      parsed_path = Hash.new
+      key_chain = Array.new
+
+      keys.each do |key|
+        if key_chain.empty?
+          parsed_path&.store(key, (key == keys.last) ? operation["value"] : {})
+        else
+          parsed_path.dig(*key_chain)&.store(key, (key == keys.last) ? operation["value"] : {})
+        end
+
+        key_chain << key
+      end
+
+      parsed_path
```

Comment:
A neat trick you can do is to use [`#reduce`](https://apidock.com/ruby/Enumerable/reduce) to set up results like this without needing an each loop to modify a var. It can be a little cleaner.

```suggestion
      keys.each_with_index.reduce({}) do |acc, (key, index)|
        value = key == keys.last ? operation["value"] : {}

        if index == 0
          acc.store(key, value)
        else
         key_path = keys.slice(0..(index - 1)) # Get the path except for the current key
         acc.dig(*key_path)&.store(key,  value)
        end
      end
```